warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IvaratorRequiredVisitor.java

```diff
@@ -4,13 +4,17 @@
 import datawave.query.jexl.nodes.ExceededValueThresholdMarkerJexlNode;
 import datawave.query.jexl.nodes.QueryPropertyMarker;
 import org.apache.commons.jexl2.parser.ASTAndNode;
+import org.apache.commons.jexl2.parser.ASTFunctionNode;
+import org.apache.commons.jexl2.parser.ASTGENode;
+import org.apache.commons.jexl2.parser.ASTGTNode;
+import org.apache.commons.jexl2.parser.ASTLENode;
+import org.apache.commons.jexl2.parser.ASTLTNode;
 import org.apache.commons.jexl2.parser.JexlNode;
 
 /**
  * A visitor that checks the query tree to determine if the query requires an ivarator (ExceededValue or ExceededOr)
- * 
  */
-public class IvaratorRequiredVisitor extends BaseVisitor {
+public class IvaratorRequiredVisitor extends ShortCircuitBaseVisitor {
     
     private boolean ivaratorRequired = false;
```

Comment:
There is a way to short circuit this visitor even faster. If the boolean `ivaratorRequired` is set to true there is no need to continue visiting the rest of the tree.

For take the method that visits an AND node for example
```
    @Override
    public Object visit(ASTAndNode node, Object data) {
        node.childrenAccept(this, data);
        return data;
    }


# becomes

    @Override
    public Object visit(ASTAndNode node, Object data) {
        if (!ivaratorRequired) {
            node.childrenAccept(this, data);
        }
        return data;
    }
```